libnavui-lifecycle/src/main/java/com/mapbox/navigation/ui/lifecycle/internal/MapboxNavigationOwner.kt

```diff
@@ -0,0 +1,73 @@
+package com.mapbox.navigation.ui.lifecycle.internal
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.MapboxNavigationProvider
+import com.mapbox.navigation.ui.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.utils.internal.logI
+import java.util.concurrent.CopyOnWriteArraySet
+
+@ExperimentalPreviewMapboxNavigationAPI
+internal class MapboxNavigationOwner {
+    internal lateinit var navigationOptions: NavigationOptions
+        private set
+
+    private val services = CopyOnWriteArraySet<MapboxNavigationObserver>()
+    private var mapboxNavigation: MapboxNavigation? = null
+    private var attached = false
+
+    internal val carAppLifecycleObserver = object : DefaultLifecycleObserver {
+
+        override fun onStart(owner: LifecycleOwner) {
+            logI(TAG, Message("navigation.onStart"))
+            check(!MapboxNavigationProvider.isCreated()) {
+                "MapboxNavigation should only be created by the MapboxNavigationOwner"
+            }
+            check(this@MapboxNavigationOwner::navigationOptions.isInitialized) {
+                "You must set the NavigationOptions with MapboxNavigationApp.setup"
+            }
+            val mapboxNavigation = MapboxNavigationProvider.create(navigationOptions)
+            this@MapboxNavigationOwner.mapboxNavigation = mapboxNavigation
+            attached = true
+            services.forEach { it.onAttached(mapboxNavigation) }
+        }
+
+        override fun onStop(owner: LifecycleOwner) {
+            logI(TAG, Message("navigation.onStop"))
+            attached = false
+            services.forEach { it.onDetached(mapboxNavigation!!) }
+            MapboxNavigationProvider.destroy()
```

Comment:
We're using the provider here but also storing the instance ourselves in `MapboxNavigationOwner#mapboxNavigation`. We either need to nullify that reference when we call `MapboxNavigationProvider.destroy()` or do not store the instance and use `MapboxNavigationProvider#retrieve` and work on local refs only.

Reply:
Am using `MapboxNavigationProvider.destroy()` to make it be backwards compatible. Otherwise, we could move the implementation of `MapboxNavigationApp` into `MapboxNavigationProvider`

Do you have a suggestion here?

Reply:
Didn't understand you earlier here. Suggestion was to set `mapboxNavigation = null`

That's fixed with a test covering it. Also there is a function to `MapboxNavigationApp`
``` kotlin
fun current(): MapboxNavigation? = mapboxNavigationOwner.current()
```

Found that it makes some code much simpler. Also prevents the need to store instances downstream everywhere. End of the day it is the same as `MapboxNavigationProvider#retrieve`, but with it we could deprecate `MapboxNavigationProvider` entirely.